arctic/store/_ndarray_store.py

```diff
@@ -377,11 +408,20 @@ def _do_append(self, collection, version, symbol, item, previous_version, dirty_
                 segment = {'data': Binary(data), 'compressed': False}
                 segment['segment'] = version['up_to'] - 1
                 try:
-                    collection.update_one({'symbol': symbol,
-                                           'sha': checksum(symbol, segment)},
-                                          {'$set': segment,
-                                           '$addToSet': {'parent': version['base_version_id']}},
-                                          upsert=True)
+                    if ARCTIC_FORWARD_POINTERS is FwPointersCfg.DISABLED:
```

Comment:
Is this block clearer if written as two optional write operations?, the second collection.update_one is rewritten but essentially the same

Reply:
I decided to favor some code duplication in favor of minimizing the regression risks, and keep the existing code path (with disabled fw pointers) as similar as possible.

We can properly refactor this once we build confidence in our implementation.
I am adding a todo with your suggestion though.